packages/cli/lib/commands/upgrade.ts

```diff
@@ -0,0 +1,77 @@
+import { GoogleAnalytics, ProjectConfig, Util } from "@igniteui/cli-core";
+import { TemplateManager } from "../TemplateManager";
+
+let command: {
+	[name: string]: any,
+	templateManager: TemplateManager,
+	execute: (argv: any) => Promise<void>,
+	upgrade: (argv: any) => Promise<void>
+};
+
+// tslint:disable:object-literal-sort-keys
+command = {
+	command: "upgrade-packages",
+	desc: "upgrades Ignite UI Packages",
+	templateManager: null,
+	builder: {
+		"skip-install": {
+			alias: "si",
+			default: false,
+			describe: "Runs upgrade command without performing install",
+			type: "boolean"
+		}
+	},
+	async execute(argv) {
+
+		GoogleAnalytics.post({
+			t: "screenview",
+			cd: "Upgrade packages"
+		});
+
+		return command.upgrade(argv);
+	},
+	async upgrade(argv) {
+		const config = ProjectConfig.getConfig();
+		const framework = config.project.framework;
+		const projectType = config.project.projectType;
+
+		switch (framework.toLowerCase()) {
+			case "jquery":
+				Util.log("Upgrading packages for jQuery projects is currently not supported!");
+				return;
+			case "react":
+				Util.log("Upgrading packages for React projects is currently not supported!");
+				return;
+			case "angular":
+				if (projectType === "igx-ts") {
+					const projectLibrary = command.templateManager.getProjectLibrary(framework, projectType);
+					let project;
+					if (!config.project.projectTemplate) {
+						// in case project template is missing from the config we provide backward.
+						project = projectLibrary.getProject(projectLibrary.projectIds[0]);
+					} else {
+						project = projectLibrary.getProject(config.project.projectTemplate);
+					}
+					const success = await project.upgradeIgniteUIPackages(process.cwd(), "");
+					if (success && !argv.skipInstall) {
+						Util.execSync("npm install");
```

Comment:
install should have `--quiet` flag. Also, there should probably be some kind of log before the install. Otherwise, it just seems like your console froze.

Reply:
https://github.com/IgniteUI/igniteui-cli/issues/737
Logged a separate issue for this